src/ui/ui.ts

```diff
@@ -330,11 +515,11 @@ export class UI extends Phaser.GameObjects.Container {
       }
       showMessageAndCallback();
     } else {
-      this.getCurrentMessageHandler().showText(text, delay, callback, callbackDelay, prompt, promptDelay);
+      this.getCurrentMessageHandler()!.showText(text, delay, callback, callbackDelay, prompt, promptDelay);
     }
   }
 
-  showDialogue(
+  public showDialogue(
```

Comment:
NIT: tsdoc

Reply:
The whole text and dialogue thing if thoroughly undocumented and has a ton of parameters which are extremely confusing. This function is basically a buffer for calling `showDialogue` on the currently active MessageUiHandler, but that's as much as I can tell considering MessageUiHandler is not documented either.
I think this will have to be for a text/dialogue refactor, because it is a whole mess, sorry

Reply:
How about just leveraging AI to write some stuff based on what's going on in there?

```suggestion
  /**
   * Displays a dialogue box with the given text and character name, handling localization, message paging, 
   * and dialogue-skipping logic. If the text contains `$`, it will be split into multiple dialogue pages.
   * If the keyOrText matches a translation key, the localized text will be used and optionally skipped if already seen.
   *
   * @param keyOrText - The dialogue text to display, or a localization key.
   * @param name - The name of the speaking character. If undefined, no name is shown.
   * @param delay - Optional. The delay before showing the dialogue, in milliseconds.
   * @param callback - Function to call after the dialogue (and all pages, if any) has been shown.
   * @param callbackDelay - Optional. Delay before calling the callback, in milliseconds.
   * @param promptDelay - Optional. Delay before showing the prompt to continue, in milliseconds.
   */
  public showDialogue(
```

Reply:
Bad/misleading docs are worse than no docs. Those docs would have to be investigated/verified, and this isn't even a function being modified by the PR.